photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,282 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Module which provides classes to perform PSF Photometry"""
+
+from __future__ import division
+import abc, six
+import numpy as np
+from astropy.table import Table, vstack
+from photutils.psf import subtract_psf
+
+
+__all__ = ['PSFPhotometryBase', 'NStarPSFPhotometry']
+
+
+class PSFPhotometryBase(object):
+    __metaclass__ = abc.ABCMeta
+
+    @abc.abstractmethod
+    def do_photometry(self):
+        pass
+
+
+class NStarPSFPhotometry(PSFPhotometryBase):
+    """
```

Comment:
I suggest "StetsonPSFPhotometry" or "DaophotPhotometry". I think those names are better known. Also, the docs say "This class basically implements the loop FIND, GROUP, NSTAR, SUBTRACT, FIND", so it does more than just NSTAR and we certainly don't want to call it "FINDGROUPNSTARSUBTRACTFINDLoopPSFPhotometry"  ;-)